### ISSUE TITLE
SNOW-1337591: Reporting of config.toml parsing errors without debug

### ISSUE THREAD

**Issue by sfc-gh-vtimofeenko** [snowflakedb/snowflake-cli#996]
### Description

If `snow` fails to parse the config file because of toml syntax error -- it would be useful to the user to see the exact error without using `--debug` and sifting through the debug output.

Perhaps messages from `tomlkit.exceptions.*` could be directly shown to the user?

### Context

Migrating from `snowsql`'s config files which are untyped key-values nested under `[connections]` keys:

```conf
[connections.foo]
username = bar
```
If copypasted directly into `~/Library/Application\ Support/snowflake/config.toml`, an unclear error is produced:

```
❯ snow connection test                            

An unexpected exception occurred. Use --debug option to see the traceback. Exception message:

An unknown error happened while loading '/Users/vtimofeenko/Library/Application Support/snowflake/config.toml'
```

`--debug` does help, but is very chatty:

```
❯ snow connection test --debug 2>&1               

# 78 lines here

UnexpectedCharError: Unexpected character: 'b' at line 2 col 11

# 130 lines here

ConfigSourceError: An unknown error happened while loading '/Users/vtimofeenko/Library/Application Support/snowflake/config.toml'
```

**Comment by sfc-gh-turbaszek** [comment]
#857

**Comment by sfc-gh-jsikorski** [comment]
We are resolving the problem by providing more meaningful error messages (with the PR linked above)

**Comment by sfc-gh-turbaszek** [comment]
Fixed in #857
